src/config.jl

```diff
@@ -55,6 +99,18 @@ Base.eltype(cfg::AbstractConfig) = eltype(typeof(cfg))
 
 @inline (chunksize(::AbstractConfig{N})::Int) where {N} = N
 
+function maketag(kind::Union{Symbol,Nothing}, f, X)
+    if kind === :default
+        return Tag(f, X)
+    elseif kind === :small
+        return SmallTag(f, X)
```

Comment:
This is type-unstable, can we do something about it? Perhaps pass the symbol as a `Val`?

Reply:
One of my objections to this PR is that I don't much like adding keyword `tag::Union{Symbol,Nothing} = :default` to everything. (1) it's messy, and I'm not sure how often anyone needs to pass this, (2) accepting a symbol from a special list of 2 seems like an odd interface. I'm not sure `Val(:default)` is much better. (3) it's called `tag` but isn't the tag,  doesn't accept a`Tag`.

Maybe a better interface would be to use Preferences.jl, as for NaN-safe mode?

Reply:
It is called with `@inline` to avoid that instability: https://github.com/JuliaDiff/ForwardDiff.jl/pull/748/files/d88bf58768e127e9a97a642deadbcd45163662e1#diff-aeb036d3acc3ca0b52368fdcc3dbc8bca465b98bc269a1a5b4668b1dcb2e2f7eR145

I'll move it to the definition so that there's no ambiguity